DSCResources/MSFT_xIisMimeTypeMapping/MSFT_xIisMimeTypeMapping.psm1

```diff
@@ -0,0 +1,168 @@
+######################################################################################
+# DSC Resource for IIS Server level MIME Type mappings
+######################################################################################
+data LocalizedData
+{
+    # culture="en-US"
+    ConvertFrom-StringData @'
+NoWebAdministrationModule=Please ensure that WebAdministration module is installed.
+AddingType=Adding MIMEType '{0}' for extension '{1}'
+RemovingType=Removing MIMEType '{0}' for extension '{1}'
+TypeExists=MIMEType '{0}' for extension '{1}' already exist
+TypeNotPresent=MIMEType '{0}' for extension '{1}' is not present as requested
+TypeStatusUnknown=MIMEType '{0}' for extension '{1}' is is an unknown status
+'@
+}
+
+######################################################################################
+# The Get-TargetResource cmdlet.
+# This function will get the Mime type for a file extension
+######################################################################################
+function Get-TargetResource
+{
+    [OutputType([Hashtable])]
+	param
+	(		
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+		[String]$Extension,
+
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+        [String]$MimeType,
+
+        [ValidateSet("Present", "Absent")]
+        [Parameter(Mandatory)]
+        [string]$Ensure
+	)
+	
+    # Check if WebAdministration module is present for IIS cmdlets
+    CheckIISPoshModule
+
+    $getTargetResourceResult = $null;
+
+    $mt = GetMapping $Extension $MimeType 
+
+    if ($mt -eq $null)
+    {
+        $ensureResult = "Absent";
```

Comment:
but the `$getTargetResourceResult` should have a hashtable with the existing data, right? If `$mt` is null, the mapping does not exist, so we shouldn't return anything just because we happen to know both values that should be in the hash-table. We could return the hash-table with empty strings rather than a $null value?